tokio/src/io/read_buf.rs

```diff
@@ -312,15 +314,47 @@ impl fmt::Debug for ReadBuf<'_> {
 }
 
 unsafe fn slice_to_uninit_mut(slice: &mut [u8]) -> &mut [MaybeUninit<u8>] {
-    &mut *(slice as *mut [u8] as *mut [MaybeUninit<u8>])
+    // SAFETY:
+    //
+    // 1. `MaybeUninit<u8>` has the same memory layout as `u8`.
+    // 2. `*mut [u8] as *mut [MaybeUninit<u8>]` follows the
+    //    [Pointer-to-pointer cast].
+    //
+    // [Pointer-to-pointer cast]:
+    // https://doc.rust-lang.org/1.90.0/reference/expressions/operator-expr.html#r-expr.as.pointer
+    unsafe { &mut *(slice as *mut [u8] as *mut [MaybeUninit<u8>]) }
```

Comment:
You have a lot of comments like this one. I would suggest shortening/simplifying them.
```suggestion
    // SAFETY: `MaybeUninit<u8>` has the same memory layout as u8, and the caller
    // promises to not write uninit bytes to the slice.
    unsafe { &mut *(slice as *mut [u8] as *mut [MaybeUninit<u8>]) }
```